api/correlation/context.go

```diff
@@ -24,21 +24,21 @@ type correlationsType struct{}
 
 var correlationsKey = &correlationsType{}
 
-// WithMap returns a context with the Map entered into it.
-func WithMap(ctx context.Context, m Map) context.Context {
+// ContextWithMap returns a context with the Map entered into it.
+func ContextWithMap(ctx context.Context, m Map) context.Context {
 	return context.WithValue(ctx, correlationsKey, m)
 }
 
 // NewContext returns a context with the map from passed context
 // updated with the passed key-value pairs.
 func NewContext(ctx context.Context, keyvalues ...core.KeyValue) context.Context {
```

Comment:
Shouldn't this be `ContextWithKeyValues`,  `ContextWithLabels` or something like that?

Reply:
I didn't have a good idea for a relatively short-and-sweet name of this function. Your proposals are making the `ContextWithFoo` inconsistent, because what `NewContext` is doing is creating a copy of a map in the context, adding keys to it and putting it back into context. Other functions basically override whatever context had with whatever was passed to the function.

`ContextWithUpdate`? Maybe replace `keyvalues ...core.Keyvalue` with `update MapUpdate`?

Reply:
I see the point, I like the `ContextWithUpdate` although not sure about replacing the keyvalues with an Update. It makes the API more hard to use. 

If that feels strange I'm okay on doing this another time :man_shrugging:

Reply:
Yeah, tried that replacing keyvalues with update shortly. Even added some convenience constructor for add-only MapUpdate, but that was still annoying to use:

```go
ctx = correlation.ContextWithUpdate(ctx, correlation.NewMapAddUpdate(kv1, kv2, kv3))
```

Meh.

So yeah, maybe some other time in another PR.

Just got another idea for that other PR - drop the function and add `ApplyAdd` to `Map`:

```go
ctx := …
cc := correlation.MapFromContext(ctx).ApplyAdd(kv1, kv2, kv3)
ctx = correlation.ContextWithMap(ctx, cc)
// or
cc := correlation.MapFromContext(ctx)
ctx = correlation.ContextWithMap(ctx, cc.ApplyAdd(kv1, kv2, kv3)
```

Reply:
Maybe for this next PR we could make the `cc := correlation.MapFromContext(ctx)` inside the `AddKey` for example:

```go
    ctx := correlation.AddKeyValue(ctx, kv1) // AddKeyValue would get the map from the ctx argument and update it.
```

That seems very easy-to-use in a first glance but there could be a reason not to do that. Anyway, just defer to another time :)